### ISSUE TITLE
Missing CPEs for Spring Framework 3.x libraries in Dependency-Check 4.0.1 and 4.0.2

### ISSUE THREAD

**Issue by smarkwal** [dependency-check/DependencyCheck#1816]
Dear dependency-check team,

This is not (yet) a bug report, but a question to clarify whether it is worth to open a bug report at all.

In dependency-check versions 4.0.1 and 4.0.2, dependency-check does not detect all CPEs for Spring Framework 3.x libraries. I tested with spring-core-3.0.7.RELEASE.jar and spring-core-3.2.17.RELEASE.jar.

Results with dependency-check versions 3.1.2, 3.2.1, 3.3.4, and 4.0.0:
spring-core-3.0.7.RELEASE.jar: 4 CPE, 12 CVE
spring-core-3.2.17.RELEASE.jar: 3 CPE, 5 CVE

Results with dependency-check versions 4.0.1 and 4.0.2:
spring-core-3.0.7.RELEASE.jar: 2 CPE (-2), 5 CVE (-7)
spring-core-3.2.17.RELEASE.jar: 1 CPE (-2), 0 CVE (-5)

Now, before opening a bug report, I also tested with 5.0.0-M1 and 5.0.0-M2. And it seems as if the issue has been fixed (the results are even better!):

Results with dependency-check versions 5.0.0-M1 and 5.0.0-M2:
spring-core-3.0.7.RELEASE.jar: 3 CPE, 12 CVE
spring-core-3.2.17.RELEASE.jar: 2 CPE, 6 CVE

Do you plan to release a 4.0.3 version and want me to open a bug report with more details, or do you prefer to invest all time in finishing 5.0.0?

Cheers,
Stephan

**Comment by javapapo** [comment]
I have a similar issue  unfortunately 

`Failed to execute goal org.owasp:dependency-check-maven:4.0.2:check (analyze-compile) on project iccp-backend-demo: One or more exceptions occurred during dependency-check analysis: One or more exceptions occurred during dependency-check analysis
[ERROR]         The download was interrupted; unable to complete the update
[ERROR]         org.owasp.dependencycheck.data.nvdcve.DatabaseException: Exception retrieving vulnerability for cpe:/a:pivotal_software:spring_boot:2.1.3
[ERROR]         org.owasp.dependencycheck.data.nvdcve.DatabaseException: Exception retrieving vulnerability for cpe:/a:pivotal_software:spring_boot:2.1.3
`

**Comment by jeremylong** [comment]
@smarkwal all time is being spent on 5.0.0.  I have one more integration to complete and then final cleanup and testing.  The 4.0.0 branch is garbage and I do not recommend using it - WAY too many false positives and false negatives due to the change in upgrading Lucene but not updating how matching and scores were handled.

**Comment by smarkwal** [comment]
Hi @jeremylong 
Sounds good to me. I have been testing a lot with 5.0.0-M2 recently and it looks very promising. Some false negatives I had to fix with custom hints, but all in all I love the new version. I will close this issue.

Hi @javapapo
I don't think that the problem you have reported ("Exception retrieving vulnerability ...") is similar to what I have reported. Your problem seems to be caused by a Java exception thrown during execution. If you keep having this error, I suggest to open a new issue and add a stack trace and a log file.
